### DISCUSSION TITLE
vsg-qt Shortcut key exception

### DISCUSSION THREAD

**Discussion by Idealist-dreamer** [vsg-dev/VulkanSceneGraph#1341] [General]
Hi Robert,

I am using vsg-qt and I want to detect window keystroke events. For this purpose I used vsg::Visitor and overwrote void apply(vsg::KeyPressEvent& e), but I found that it did not work properly.

KeyEvent contains the following key fields.
` KeySymbol keyBase = {};
        KeySymbol keyModified = {};
        KeyModifier keyModifier = {};`

I don't know the purpose of keyModified, but I want to check for r in keyBase and ctrl and alt in keyModifier after pressing ctrl alt r. That's what I did, but I found that when I pressed ctrl alt, keyModifier was 0.

In order to track down cause I have found the VSG - qt - KeyboardMap. CPP, reading KeyboardMap: : getKeySymbol function. I notice that you take the information from QKeyEvent and update vsg::KeySymbol& keySymbol, vsg::KeySymbol& modifiedKeySymbol, vsg::KeyModifier& keyModifier.

The update process is as follows.
![vsg-qt快捷键](https://github.com/user-attachments/assets/0784a8e7-3c9e-4897-9cdd-a3e9c467acec)

The modifier keys of QKeyEvent are mixed using the | operator, so after pressing ctrl alt, e->modifiers() is Qt::ControlModifier | Qt::AltModifier. The above code will not hit ctrl or alt. Instead, enter Qt::NoModifier.

I think the following is true, don't you think?
![vsg-qt修改后](https://github.com/user-attachments/assets/fbd6ec15-e9fe-4e4c-8c7d-81bcf8bfa20c)

If this is correct, I hope to see it in future updates to vsg-qt.

**Comment by robertosfield** [comment]
Your suggest fix does look appropriate.  Could you create a PR on the vsgQt project with this fix so I can test/merge it.  Thanks.

**Reply by Idealist-dreamer** [reply]
Ok, I have submitted it to vsg-qt.

**Reply by robertosfield** [reply]
Thanks, all looks good so now merged.
